x/axelarnet/types/types.go

```diff
@@ -325,6 +327,16 @@ func (minDeposits CallContractProposalMinDeposits) ToMap() callContractProposalM
 		}
 
 		minDepositsMap[chain][contractAddress] = minDeposit.MinDeposits
+
+		// TODO: eventually, this is confusing and bad cuz cosmos addresses will also
+		// show up here and be prefixed with 0x. Like the address validator, we should
+		// also implement chain-specific address deserializer so that we just use the
+		// actual bytes as map keys for this check instead of a string representation.
+		if strings.HasPrefix(contractAddress, ZERO_X_PREFIX) {
```

Comment:
can we check if it's an evm chain? maybe pass in the nexus keeper or refactor it more? Feels weird to ship it like this

Reply:
This is precisely what the audit suggests. I know what you mean; the refactor isn't as easy as the TODO I wrote. Since this is only used by EVM for now, I say we ship and come back to it.